tools/cldr-code/src/test/java/org/unicode/cldr/unittest/TestCheckWidths.java

```diff
@@ -40,4 +42,26 @@ public void TestBasic() {
             assertEquals("errorSubType", Subtype.valueTooWide, err.getSubtype());
         }
     }
+
+    public void TestEnglishWidths() {
+        UnitWidthUtil unitWidthUtils = CheckWidths.UnitWidthUtil.getInstance();
+        CLDRConfig config = CLDRConfig.getInstance();
+        CLDRFile english = config.getEnglish();
+        CheckWidths x = new CheckWidths();
+        Options options = new Options();
+        List<CheckStatus> possibleErrors = new ArrayList<>();
+        x.setCldrFileToCheck(english, options, possibleErrors);
+        assertEquals("setCLDRFile", Collections.emptyList(), possibleErrors);
+        for (String path : config.getEnglish().fullIterable()) {
+            String fullPath = english.getFullXPath(path);
+            String value = english.getStringValue(path);
+            x.check(path, fullPath, value, options, possibleErrors);
+            if (!assertEquals(path + " = " + value, Collections.emptyList(), possibleErrors) || isVerbose()) {
+                // for debug
+                double roughMax = unitWidthUtils.getRoughComponentMax(path);
```

Comment:
ideally we'd have some specific API level (instead of check-level) tests here, especially checking `metric-cup`, `100-…` etc.. could be future.